pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/EqualsPredicateEvaluatorFactory.java

```diff
@@ -72,6 +72,7 @@ public static BaseRawValueBasedPredicateEvaluator newRawValueBasedEvaluator(EqPr
       case TIMESTAMP:
         return new LongRawValueBasedEqPredicateEvaluator(TimestampUtils.toMillisSinceEpoch(value));
       case STRING:
+      case JSON:
```

Comment:
Not sure if we want to allow EQ operation on `JSON` field, same for other predicates

Reply:
Reverting all PredicateEvaluator changes. We may need to support EQUALS and NOT_EQUALS, but it doesn't need to be done with this PR.